File: 2016/gara2/robot/att/robot.pas
const
    MAXN = 512;
type matrix = array[0..MAXN-1] of array[0..MAXN-1] of char;

function osserva(N, M: longint; var T: matrix): longint;
begin
    (* Mettete qui il codice della soluzione *)
    if T[0][0] = 'X' then osserva := -1
    else osserva := 42;
end;


var
    N, M, i : longint;
    T       : matrix;
    fr, fw  : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, M);
    for i:=0 to N-1 do
        readln(fr, T[i]);

    writeln(fw, osserva(N, M, T));
    close(fr);
    close(fw);
end.

File: ed2/gara2/robot/att/robot.c
#include <stdio.h>
#include <assert.h>

#define MAXN 512

typedef char matrix[MAXN][MAXN];

int osserva(int N, int M, matrix T) {
    // Mettete qui il codice della soluzione
    if (T[0][0] == 'X')
        return -1;
    return 42;
}


int main() {
    FILE *fr, *fw;
    int N, M, i;
    matrix T;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d %d", &N, &M));
    for(i=0; i<N; i++)
        assert(1 == fscanf(fr, "%s", T[i]));
    fprintf(fw, "%d\n", osserva(N, M, T));
    fclose(fr);
    fclose(fw);
    return 0;
}
